ddtrace/contrib/vertica/patch.py

```diff
@@ -0,0 +1,246 @@
+import importlib
+import wrapt
+
+import ddtrace
+from ddtrace import config, Pin
+from ddtrace.ext import net, AppTypes
+from ddtrace.utils.wrappers import unwrap
+
+from .constants import APP
+from ...ext import db as dbx
+
+"""
+Note: we DO NOT import the library to be patched at all!
+
+What this means is that this approach completely solves our patching problem as
+well. wrapt will hook into the module import system for the patches that we
+provide.
+"""
+
+_PATCHED = False
+
+def copy_before(instance, span, conf, *args, **kwargs):
```

Comment:
Just thinking out loud. What if we replace `def copy_before():` and `def execute_before()` with a function `def tag_query()` and then in the configuration:

```
...
  "execute": {
    "on_before": tag_query
  },
  "copy": {
    "on_before": tag_query
  }
...
```
In the future (now seems to be not necessary), "on_before" etc... could also be passed a list of operations, instead of only one.

This adds a few benefits IMO:
 - greater direction between the function name and what it does
 - we do not lose the relation between the "on_before" and the fact that a tag is set
 - less code repetition

This also apply to `execute_after` and `fetch_after`.